tests/tensor/rewriting/test_basic.py

```diff
@@ -1349,46 +1349,26 @@ def has_no_join(fgraph_ops):
 
 
 def test_local_join_empty():
-    # Vector case - empty tensors should be removed and join optimized
+    # Vector case - empty tensors should be removed
```

Comment:
I now realize why you were changing this test. Maybe change the test case to be `pt.join(0, vec, vec[::-1], empty_ver)`? Then you can leave the test almost untouched except for the expected graph values.

Calling `rewrite_graph` on the expected graph isn't a strong test, because we don't know what it is doing. It could be rendering the two graphs equivalent but wrong